cardano-cli/src/Cardano/CLI/Shelley/Run/StakeAddress.hs

```diff
@@ -10,23 +10,52 @@ import qualified Data.ByteString.Lazy.Char8 as LBS
 import qualified Data.Text.IO as Text
 
 import           Control.Monad.Trans.Except (ExceptT)
-import           Control.Monad.Trans.Except.Extra (firstExceptT, newExceptT)
+import           Control.Monad.Trans.Except.Extra (firstExceptT, hoistEither, left, newExceptT)
 
 import           Cardano.Api
 
 import           Cardano.Api (StakingVerificationKey (..),
                    readStakingVerificationKey,
                    shelleyDeregisterStakingAddress, shelleyDelegateStake,
                    shelleyRegisterStakingAddress, writeCertificate)
-import           Shelley.Spec.Ledger.Keys (hashKey)
+import           Shelley.Spec.Ledger.Keys (VKey(..), hashKey)
 import           Cardano.Config.Shelley.ColdKeys hiding (writeSigningKey)
+import qualified Cardano.Crypto.DSIGN as DSIGN
+
 
 import           Cardano.CLI.Helpers (textToLByteString)
 import           Cardano.CLI.Shelley.Parsers
 
 data ShelleyStakeAddressCmdError
   = ShelleyStakeReadPoolOperatorKeyError !FilePath !KeyError
+  | ShelleyStakeAddressConvError !ConversionError
+  | ShelleyStakeAddressKeyPairError
+      !Text
+      -- ^ bech32 private key
+      !Text
+      -- ^ bech32 public key
+  | ShelleyStakeAddressReadFileErr !FilePath !Text
```

Comment:
For consistency, I would call this

```suggestion
  | ShelleyStakeAddressReadFileError !FilePath !Text
```

and also add the `Error` suffix to the new error constructors added below.